urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/businesslogic/BookingBL.java

```diff
@@ -0,0 +1,109 @@
+package com.moblima.businesslogic;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import com.moblima.dataaccess.BookingDAO;
+import com.moblima.dataaccess.BookingDAOImpl;
+import com.moblima.dataaccess.SeatDAO;
+import com.moblima.dataaccess.SeatDAOImpl;
+import com.moblima.model.Booking;
+import com.moblima.model.BookingStatus;
+import com.moblima.model.Cinema;
+import com.moblima.model.Movie;
+import com.moblima.model.MovieGoer;
+import com.moblima.model.MovieTicket;
+import com.moblima.model.Seat;
+import com.moblima.model.ShowTime;
+import com.moblima.model.TicketType;
+import com.moblima.model.Transaction;
+import com.moblima.model.TransactionMethod;
+
+
+public class BookingBL{
+	private static float bookingPrice = 1.5f;
+	
+	private BookingDAO bookingDAO;
+	private SeatDAO seatDAO;
+	private TransactionBL transactionBL;
+	private MovieGoerBL movieGoerBL;
+	private MovieTicketBL movieTicketBL;
+	private TicketPriceBL ticketPriceBL;
+
+	public BookingBL(){
+		bookingDAO = BookingDAOImpl.getInstance();
+		seatDAO = SeatDAOImpl.getInstance();
+		transactionBL = new TransactionBL();
+		movieGoerBL = new MovieGoerBL();
+		movieTicketBL = new MovieTicketBL();
+		ticketPriceBL = new TicketPriceBL();
+	}
+
+	public Booking createBooking(MovieGoer movieGoer, ShowTime showTime, TransactionMethod transactionMethod, String[] seats){
+		Transaction transaction;
+		List<MovieTicket> movieTickets;
+		MovieTicket movieTicket;
+		
+		Cinema cinema = showTime.getCinema();
+		Date now = Calendar.getInstance().getTime();
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmm");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.